internal/job/job.go

```diff
@@ -154,78 +166,72 @@ func (t *Job) GetGroupJobState(name string, groupUUID string) (*GroupJobState, e
 		return nil, errors.New("empty group")
 	}
 
-	jobStates := make([]jobState, 0, len(taskStates))
-	for _, taskState := range taskStates {
-		var results []any
-		for _, result := range taskState.Results {
-			switch name {
-			case PreheatJob:
-				var resp PreheatResponse
-				if err := UnmarshalTaskResult(result.Value, &resp); err != nil {
-					return nil, err
-				}
-				results = append(results, resp)
-			case GetTaskJob:
-				var resp GetTaskResponse
-				if err := UnmarshalTaskResult(result.Value, &resp); err != nil {
-					return nil, err
+	jobStates := make([]jobState, len(taskStates))
+	var (
+		wg       sync.WaitGroup
```

Comment:
Why not use `https://pkg.go.dev/golang.org/x/sync/errgroup` for easier goroutine and error management?